### PR TITLE
do not automatically enable xformers

### PR DESCRIPTION
**Pull request by patrickvonplaten**

### CODE REVIEW

File: src/diffusers/models/attention.py [hunk 1]
```diff
@@ -447,16 +446,6 @@ def __init__(
         # 3. Feed-forward
         self.norm3 = nn.LayerNorm(dim)
 
-        # if xformers is installed try to use memory_efficient_attention by default
```
Comment (patrickvonplaten) [inline src/diffusers/models/attention.py:450]: We should not automatically enable xformers for two reasons:

- 1. It's not PyTorch-like API. PyTorch doesn't by default enable all the fastest options available
- 2. We allocate GPU memory before the user even does `.to("cuda")` 
- 3. This behavior is not consistent with cases where xformers is not installed

Comment (patrickvonplaten) [reply src/diffusers/models/attention.py:450]: cc @pcuenca @anton-l @patil-suraj

**Review by anton-l — APPROVED** [review]
Very much agree with the reasoning, thanks for the update!

File: src/diffusers/models/attention.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/diffusers/models/attention.py:450]: Related issues: https://github.com/huggingface/diffusers/issues/1565
https://github.com/huggingface/diffusers/issues/1344#issuecomment-1335017683

**Pull request merged**